hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,343 @@
+#!/bin/bash
+
+conf_file=""
+
+declare -a netnses
+
+declare -A pfs
+declare -A pcis
+declare -A pf_port_names
+declare -A pf_switch_ids
+
+TIMEOUT="${TIMEOUT:-2}"
+POLL_INTERVAL="1"
+
+while test $# -gt 0; do
+  case "$1" in
+
+   --netns | -n)
+      netnses+=("$2")
+      shift
+      shift
+      ;;
+
+   --pf | -d)
+      if [[ -z "${netnses[-1]}" ]];then
+          echo "ERROR: No netns is specified, please provide one using the --netns before using --pf."
+          exit 1
+      fi
+
+      pfs["${netnses[-1]}"]+="$2 "
+      shift
+      shift
+      ;;
+
+   --conf-file | -c)
+      conf_file=$2
+      shift
+      shift
+      ;;
+
+   --help | -h)
+      echo "
+vf-netns-switcher.sh --netns <> --pf <> [--conf-file <>]:
+
+	--netns | -n		Netns to switch the interface PFs and VFs to. This needs to be provided before the pfs
+				to associate the pfs to the netns. This flag can be repeated.
+
+	--pf | -d		The PF to switch it and its VFs to the specified netns. This must be provided after a
+				netns to associate the pf to the last provided netns. This option can be repeated.
+
+	--conf-file | -c	A file to read confs from, this will override cli flags. Conf file should be of the form:
+				- netns: <netns1>
+				  pfs:
+				  - <pf1>
+				  - <pf2>
+				- netns: <netns2>
+				  pfs:
+				  - <pf3>
+				  - <pf4>
+
+"
+      exit 0
+      ;;
+
+   *)
+      echo "No such option!!"
+      echo "Exitting ...."
```

Comment:
> "No such option!!"

It reads abit aggressive :) , maybe just :

echo "Error: Invalid option, Exiting..."

Reply:
thanks, done